rotkehlchen/db/utils.py

```diff
@@ -432,6 +432,8 @@ def db_tuple_to_str(
 
     # else can only be evm transaction
     assert tuple_type == 'evm_transaction', 'only DBTupleType possible here is evm_transaction'
+    if isinstance(data[0], int):
+        return f'Optimism transaction with hash 0x{data[1].hex()} and l1_fee {data[0]}'
```

Comment:
Ok, but if the flow ends up here without it, it would raise an exception "int has no method hex()".

Reply:
The query used for optimism is different and needs the l1_fee as the first argument of the tuple. I'm not sure if the query can be changed so that tx_hash is first (or even if it's worth it). Maybe a new tuple_type makes sense since here we add the optimism_transaction in it's own table instead of a common evm_transaction?

Reply:
Didn't you change this? I saw that you changed the order as I asked, didn't you?

Reply:
Another reason this **has** to be done as I said is that you would break this logic by what you did. https://github.com/rotki/rotki/blob/a51be249f1f38cf2b28e0bf28ebdfbcb33844dcc/rotkehlchen/db/dbhandler.py#L1554

The code for relevant address detection expects either evm transaction or internal tx. So accidentally without this change all relevant address for optimism transactions would take wrong transaction hash